pandas/core/arrays/integer.py

```diff
@@ -553,6 +553,38 @@ def cmp_method(self, other):
         name = f"__{op.__name__}__"
         return set_function_name(cmp_method, name, cls)
 
+    def _accumulate(self, name: str, skipna: bool = True, **kwargs) -> "IntegerArray":
+        data = self._data
+        mask = self._mask
+
+        cum_function, fill_value = {
+            "cumprod": (np.cumprod, 1),
+            "cummax": (np.maximum.accumulate, data.min()),
+            "cumsum": (np.cumsum, 0),
+            "cummin": (np.minimum.accumulate, data.max()),
+        }[name]
+        from ..nanops import _get_values
+
+        values, mask, dtype, dtype_max, fill_value = _get_values(
+            data, skipna=skipna, fill_value=fill_value, mask=mask
+        )
+
+        if not skipna:
+            mask = np.maximum.accumulate(mask)
+
+        # makes target dtypes explicit since CI showed optimal UInt32
+        # dtype on test data occasionally. This was different across systems
+        dtype_out = dtype
+        if name in ["cumsum", "cumprod"]:
+            if dtype.name.lower().startswith("u"):
+                dtype_out = "uint64"
+            else:
+                dtype_out = "int64"
+
+        vals = cum_function(values, dtype=dtype_out)
```

Comment:
you should be able to use nanops.nan_accum_func no?

Reply:
I could use this, but see my comment on https://github.com/pandas-dev/pandas/pull/28509#discussion_r440436191
I'm happy to go in any direction here